### PR TITLE
[stateless_validation] Add helper functions for chunk endorsement validation

### PR DESCRIPTION
**Pull request by shreyan-gupta**
Couple of awesome changes here...
- Expose validator_mandates_config as we need this to calculating total stake. See function `does_chunk_have_enough_stake`
- `Vec<Option<Box<Signature>>>`??? Wth is this??? Type aliased it to `ChunkEndorsementSignatures`
- Added `validate_signature` function to `ChunkEndorsement` that checks whether a signature + public key is valid or not. This is used for verifying chunk endorsement signature in header.
- `ChunkValidatorAssignments` just got coooooler!! Now we have a `does_chunk_have_enough_stake` function that calculates whether we have 2/3rd stake and we are good to go?
- Some other minor improvements to `ChunkValidatorAssignments` like adding `ordered_chunk_validators` function and `contains` function

### CODE REVIEW

File: chain/chain/src/test_utils/kv_runtime.rs [hunk 1]
```diff
@@ -706,6 +707,13 @@ impl EpochManagerAdapter for MockEpochManager {
         Ok(chunk_producers[index].account_id().clone())
     }
 
+    fn get_validator_mandates_config(
```
Comment (robin-near) [inline chain/chain/src/test_utils/kv_runtime.rs:710]: I see your point in the PR description, but now I wonder if AssignmentWeight should just be a Balance instead of a struct of number of full seats plus a partial balance. Then we wouldn't need to multiply it again to calculate anything. What do you think?

We can do that in a separate PR.

**Review by robin-near — APPROVED** [review]

File: chain/chain/src/test_utils/kv_runtime.rs (hunk 1 above)
Comment (shreyan-gupta) [reply chain/chain/src/test_utils/kv_runtime.rs:710]: Hmm, that makes sense... Why really keep all these seat numbers and partial seats etc.... We just need the overall balance

Comment (shreyan-gupta) [reply chain/chain/src/test_utils/kv_runtime.rs:710]: Separate PR it is!
